pkg/service/model_build_target_group.go

```diff
@@ -0,0 +1,392 @@
+package service
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"regexp"
+	elbv2api "sigs.k8s.io/aws-load-balancer-controller/apis/elbv2/v1beta1"
+	"sigs.k8s.io/aws-load-balancer-controller/pkg/annotations"
+	"sigs.k8s.io/aws-load-balancer-controller/pkg/k8s"
+	elbv2model "sigs.k8s.io/aws-load-balancer-controller/pkg/model/elbv2"
+	"strconv"
+	"strings"
+)
+
+const (
+	tgAttrsProxyProtocolV2Enabled  = "proxy_protocol_v2.enabled"
+	tgAttrsPreserveClientIPEnabled = "preserve_client_ip.enabled"
+	healthCheckPortTrafficPort     = "traffic-port"
+)
+
+func (t *defaultModelBuildTask) buildTargetGroup(ctx context.Context, port corev1.ServicePort, tgProtocol elbv2model.Protocol) (*elbv2model.TargetGroup, error) {
+	svcPort := intstr.FromInt(int(port.Port))
+	tgResourceID := t.buildTargetGroupResourceID(k8s.NamespacedName(t.service), svcPort)
+	if targetGroup, exists := t.tgByResID[tgResourceID]; exists {
+		return targetGroup, nil
+	}
+	healthCheckConfig, err := t.buildTargetHealthCheckConfig(ctx)
```

Comment:
nit, should be `buildTargetGroupHealthCheckConfig`

Reply:
correct, updated.